src/ssl_stubs.c

```diff
@@ -1621,8 +1625,10 @@ CAMLprim value ocaml_ssl_write_blocking(value socket, value buffer, value start,
   char *buf = (char*)String_val(buffer) + Int_val(start);
   SSL *ssl = SSL_val(socket);
 
+  if (Int_val(start) < 0) caml_invalid_argument("Ssl.write: negative offset");
+  if (Int_val(length) < 0) caml_invalid_argument("Ssl.write: negative length");
   if (Int_val(start) + Int_val(length) > caml_string_length(buffer))
-    caml_invalid_argument("Buffer too short.");
+    caml_invalid_argument("Ssl.write: Buffer too short.");
```

Comment:
I still wonder if this should refer to the actual function name (write vs write_blocking) so that the caller knows the real source. Perhaps not that important.